rust/arrow/src/array.rs

```diff
@@ -848,6 +901,16 @@ impl BinaryArray {
         self.value_offset_at(i + 1) - self.value_offset_at(i)
     }
 
+    /// Returns a reference of the value offset buffer
```

Comment:
The comment says this is returning a reference, but the implementation is using `clone` and not returning a reference. This is confusing.

Reply:
I agree it is a little confusing. Let me change "reference" to "clone" to make it clearer.